vms/rpcchainvm/ghttp/http_test.go

```diff
@@ -84,6 +88,57 @@ func TestRequestClientArbitrarilyLongBody(t *testing.T) {
 	client.ServeHTTP(w, r) // Shouldn't block forever reading the body
 }
 
+// Tests that writes to the http response in the server are propagated to the
+// client
+func TestHttpResponse(t *testing.T) {
+	require := require.New(t)
+
+	handler := http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.Header()["Foo"] = []string{"bar"}
+		_, err := w.Write([]byte{1, 2, 3})
+		require.NoError(err)
+	})
+
+	listener, err := grpcutils.NewListener()
+	require.NoError(err)
+	server := grpc.NewServer()
+	httppb.RegisterHTTPServer(server, NewServer(handler))
+
+	go func() {
+		require.NoError(server.Serve(listener))
+	}()
+
+	conn, err := grpc.NewClient(
+		listener.Addr().String(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	require.NoError(err)
+
+	client := NewClient(httppb.NewHTTPClient(conn))
+
+	recorder := &httptest.ResponseRecorder{
+		Body: bytes.NewBuffer(nil),
+	}
+
+	request := &http.Request{
+		Body: io.NopCloser(strings.NewReader("bar")),
+		Header: http.Header{
+			"Upgrade": {"upgrade"},
+			"bar":     {"bar"},
+		},
+		Trailer: http.Header{
+			"bar": {"bar"},
+		},
+		RemoteAddr: "bar",
+	}
+
+	client.ServeHTTP(recorder, request)
+
+	require.Equal(200, recorder.Code)
+	require.Equal([]string{"bar"}, recorder.Header()["Foo"])
```

Comment:
```suggestion
	require.Equal(
		http.Header{
			"Foo": []string{"bar"},
		},
		recorder.Header(),
	)
```

Reply:
This is also nicer since it captures that bug you found....